src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.wasm.cs

```diff
@@ -347,14 +349,18 @@ private bool OnScroll(object sender, RoutedEventArgs routedEventArgs)
 			// This would however not include scrolling due to the inertia which should also be flagged as intermediate.
 			// The main issue is that the IsPointerPressed be true ONLY when dragging the scrollbars with the mouse, 
 			// as for finger and pen we will get a PointerCancelled which will reset the pressed state to false.
-			// And it would also requires us to explicitly invoke OnScroll in PointerRelease in order to raise the
-			// final SV.ViewChanged event with a IsIntermediate == false.
-			// This is probably safer for now to always consider the scroll as final, even if it introduce a performance cost
-			// (the SV updates mode is always sync when isIntermediate is false).
-			var isIntermediate = false;
-
+			// And it would also require us to explicitly invoke OnScroll in PointerRelease in order to raise the
+			// final SV.ViewChanged event with an IsIntermediate == false.
+			// As a best-effort guess, we will consider the scroll as intermediate if the native offset is different from the last ScrollTo request.
 			var horizontalOffset = GetNativeHorizontalOffset();
 			var verticalOffset = GetNativeVerticalOffset();
+			var isIntermediate =
```

Comment:
Is it possible that `_lastScrollToRequest` is "not possible" (i.e. one of the offset is out of the scrollable range)?

Reply:
I'll verify